internal/controller/isbservicerollout/isbservicerollout_controller_test.go

```diff
@@ -404,12 +404,44 @@ func Test_reconcile_isbservicerollout_Progressive(t *testing.T) {
 
 	progressiveUpgradeStrategy := apiv1.UpgradeStrategyProgressive
 
+	successfulPipelineRollout := ctlrcommon.CreateTestPipelineRollout(numaflowv1.PipelineSpec{InterStepBufferServiceName: ctlrcommon.DefaultTestISBSvcRolloutName},
+		map[string]string{}, map[string]string{}, map[string]string{}, map[string]string{},
+		&apiv1.PipelineRolloutStatus{
+			ProgressiveStatus: apiv1.PipelineProgressiveStatus{
+				UpgradingPipelineStatus: &apiv1.UpgradingPipelineStatus{
+					InterStepBufferServiceName: defaultUpgradingISBSvc.GetName(),
+					UpgradingPipelineTypeStatus: apiv1.UpgradingPipelineTypeStatus{
+						UpgradingChildStatus: apiv1.UpgradingChildStatus{
+							Name:                   defaultUpgradingPipelineName,
+							AssessmentResult:       apiv1.AssessmentResultSuccess,
+							InitializationComplete: true,
+						},
+					},
+				},
+			},
+		},
+	)
+	failedPipelineRollout := successfulPipelineRollout.DeepCopy()
+	failedPipelineRollout.Status.ProgressiveStatus.UpgradingPipelineStatus.UpgradingPipelineTypeStatus.AssessmentResult = apiv1.AssessmentResultFailure
+
+	successfulUpgradingISBServiceStatus := &apiv1.UpgradingISBServiceStatus{
+		UpgradingChildStatus: apiv1.UpgradingChildStatus{
+			Name:                   ctlrcommon.DefaultTestISBSvcRolloutName + "-1",
+			AssessmentStartTime:    &metav1.Time{Time: time.Now().Add(-1 * time.Minute)},
+			AssessmentEndTime:      &metav1.Time{Time: time.Now().Add(-30 * time.Second)},
+			AssessmentResult:       apiv1.AssessmentResultSuccess,
+			InitializationComplete: true,
+		},
+	}
+	failedUpgradingISBServiceStatus := successfulUpgradingISBServiceStatus.DeepCopy()
+	failedUpgradingISBServiceStatus.UpgradingChildStatus.AssessmentResult = apiv1.AssessmentResultFailure
```

Comment:
As part of adding tests, I decided to refactor a bit (finding some definitions that were common and creating variables out of them)